solutions/assignment1/library.py

```
@@ -0,0 +1,95 @@
+from re import search
+from sys import stderr
+
+
+class ParserError(Exception):
+
+    # Constructor or Initializer
+    def __init__(self, value):
+        self.value = value
+
+    # __str__ is to print() the value
+    def __str__(self):
+        return(str(self.value))
+
+
+class Parser:
+
+    def __init__(self):
+        # print("class created")
+        self._expected_args = {}
+        self._required_args = {}
+
+    def add_option(self, *args, **kwargs):
+        """ Function for adding expected arguments info
+            add_option(name,required=true,type=float,unique_key=local)
+        Args:
+            **kwargs : Keyword Arguments with expected arguments information
+        """
+        for arg in args:
+            self._expected_args[arg] = kwargs
+
+    def parse(self, argslist):
+        """ Function for parsing and validating the commands
+
+        Args:
+            argslist : List of commands given by user
+
+        Returns:
+            The User passed values if everything goes well , otherwise returns the error
+        """
+        # Storing required Args for Error Checking
+        required_args = set(
+            [name for name, value in self._expected_args.items() if "required" in value])
+
+        unique_key_nums = 0
+        present_unique_key = ""
+        previous_unique_key = ""
+        user_passed_args = {}
+
+        for keys in argslist:
+            if keys.startswith('--'):
```

If this isn't true, what happens? You might want to raise an exception.